server/src/main/java/org/elasticsearch/cluster/DiffableUtils.java

```diff
@@ -710,26 +703,6 @@ public Integer readKey(StreamInput in) throws IOException {
         }
     }
 
-    /**
-     * Serializes Integer keys of a map as a VInt. Requires keys to be positive.
-     */
-    private static final class VIntKeySerializer implements KeySerializer<Integer> {
-        public static final IntKeySerializer INSTANCE = new IntKeySerializer();
```

Comment:
This is the mistake - it should have been `public static final VIntKeySerializer INSTANCE = new VIntKeySerializer();`